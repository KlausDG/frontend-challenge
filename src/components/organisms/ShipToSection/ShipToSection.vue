<script setup lang="ts">
import { Badge, Card, Stack, Typography } from '@/components/atoms'
import { IconText } from '@/components/molecules'
import type { Address } from '@/models'

defineProps<{
  data: Address
}>()
</script>

<template>
  <section>
    <Card title="Ship to">
      <template #subtitle>
        <Stack :spacing="8" class="subtitle__container">
          <Typography tag="p" variant="semibold" color="dark">{{ data.name }}</Typography>
          <Badge>#{{ data.code }}</Badge>
        </Stack>
      </template>
      <Stack direction="column">
        <Stack>
          <Stack direction="column" :spacing="4" class="content__column">
            <IconText icon="map-marker" :text="data.address" />
            <IconText icon="user" :text="data.contact.name" />
          </Stack>
          <Stack direction="column" :spacing="4">
            <IconText icon="envelope" :text="data.contact.email" />
            <IconText icon="phone" :text="data.contact.phone" />
            <IconText icon="print" :text="data.contact.fax" />
          </Stack>
        </Stack>
      </Stack>
    </Card>
  </section>
</template>

<style lang="scss" scoped>
.subtitle__container {
  align-items: center;
}

.content__column {
  width: 50%;
}
@media (min-width: 1024px) {
}
</style>
