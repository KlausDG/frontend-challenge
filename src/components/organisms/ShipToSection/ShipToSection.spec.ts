import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import ShipToSection from './ShipToSection.vue'

const CardMock = {
  name: 'Card',
  props: ['title'],
  template: '<div class="mock-card"><slot name="subtitle" /><slot /></div>',
}

const StackMock = {
  name: 'Stack',
  props: ['direction', 'spacing', 'class'],
  template: '<div class="mock-stack"><slot /></div>',
}

const TypographyMock = {
  name: 'Typography',
  props: ['tag', 'variant', 'color'],
  template: '<div class="mock-typography"><slot /></div>',
}

const BadgeMock = {
  name: 'Badge',
  props: [''],
  template: '<div class="mock-badge"><slot /></div>',
}

const IconTextMock = {
  name: 'IconText',
  props: ['icon', 'text'],
  template: '<div class="mock-icon-text">{{ icon }}:{{ text }}</div>',
}

describe('ShipToSection.vue', () => {
  const addressData = {
    name: 'Jane Doe',
    code: '001',
    address: '789 Shipping St',
    contact: {
      name: 'John Smith',
      email: 'john@example.com',
      phone: '123456789',
      fax: '987654321',
    },
  }

  const wrapper = mount(ShipToSection, {
    props: { data: addressData },
    global: {
      components: {
        Card: CardMock,
        Stack: StackMock,
        Typography: TypographyMock,
        Badge: BadgeMock,
        IconText: IconTextMock,
      },
    },
  })

  it('renders Card with title "Ship to"', () => {
    const card = wrapper.findComponent(CardMock)
    expect(card.exists()).toBe(true)
    expect(card.props('title')).toBe('Ship to')
  })

  it('renders subtitle with Typography and Badge', () => {
    const typography = wrapper.findComponent(TypographyMock)
    expect(typography.exists()).toBe(true)

    const badge = wrapper.findComponent(BadgeMock)
    expect(badge.exists()).toBe(true)
    expect(badge.text()).toBe(`#${addressData.code}`)
  })

  it('renders all IconText components with correct props', () => {
    const iconTexts = wrapper.findAllComponents(IconTextMock)
    expect(iconTexts).toHaveLength(5)

    expect(iconTexts[0].props()).toEqual({ icon: 'map-marker', text: addressData.address })
    expect(iconTexts[1].props()).toEqual({ icon: 'user', text: addressData.contact.name })
    expect(iconTexts[2].props()).toEqual({ icon: 'envelope', text: addressData.contact.email })
    expect(iconTexts[3].props()).toEqual({ icon: 'phone', text: addressData.contact.phone })
    expect(iconTexts[4].props()).toEqual({ icon: 'print', text: addressData.contact.fax })
  })

  it('renders content__column class correctly', () => {
    const columnStack = wrapper.find('.content__column')
    expect(columnStack.exists()).toBe(true)
  })

  it('renders subtitle__container class correctly', () => {
    const subtitleStack = wrapper.find('.subtitle__container')
    expect(subtitleStack.exists()).toBe(true)
  })

  it('matches the component snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
