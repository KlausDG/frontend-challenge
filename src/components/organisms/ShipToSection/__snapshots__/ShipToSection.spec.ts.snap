// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShipToSection.vue > matches the component snapshot 1`] = `
"<section data-v-73c9217f="">
  <div data-v-69f42c1a="" data-v-9bb7b3b9="" data-v-73c9217f="" class="stack__container card__container" style="flex-direction: column; gap: 16px;">
    <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container" style="flex-direction: column; gap: 4px;">
      <p data-v-9bb7b3b9="" class="typography _caption_d0cbd2 _text_d0cbd2">Ship to</p>
      <div data-v-69f42c1a="" data-v-73c9217f="" class="stack__container subtitle__container" style="flex-direction: row; gap: 8px;">
        <p data-v-73c9217f="" class="typography _semibold_d0cbd2 _dark_d0cbd2">Jane Doe</p>
        <div data-v-df2d503b="" data-v-73c9217f="" class="badge__container">
          <p data-v-df2d503b="" class="typography _caption_d0cbd2 _primary_d0cbd2">#001</p>
        </div>
      </div>
    </div>
    <div data-v-69f42c1a="" data-v-73c9217f="" class="stack__container" style="flex-direction: column; gap: 0px;">
      <div data-v-69f42c1a="" data-v-73c9217f="" class="stack__container" style="flex-direction: row; gap: 0px;">
        <div data-v-69f42c1a="" data-v-73c9217f="" class="stack__container content__column" style="flex-direction: column; gap: 4px;">
          <div data-v-70c01fa4="" data-v-73c9217f="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="map-marker"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">789 Shipping St</p>
          </div>
          <div data-v-70c01fa4="" data-v-73c9217f="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="user"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">John Smith</p>
          </div>
        </div>
        <div data-v-69f42c1a="" data-v-73c9217f="" class="stack__container" style="flex-direction: column; gap: 4px;">
          <div data-v-70c01fa4="" data-v-73c9217f="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="envelope"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">john@example.com</p>
          </div>
          <div data-v-70c01fa4="" data-v-73c9217f="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="phone"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123456789</p>
          </div>
          <div data-v-70c01fa4="" data-v-73c9217f="" class="icon-text__container">
            <unicon data-v-70c01fa4="" name="print"></unicon>
            <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">987654321</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
